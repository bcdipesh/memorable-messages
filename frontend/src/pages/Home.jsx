import { useContext, useEffect } from "react";
import { Link } from "react-router-dom";

import AuthContext from "@/contexts/authContext/AuthContext";

import hero from "@/assets/hero.svg";
import { Button } from "@/components/ui/button";

const Home = () => {
  const { token, username } = useContext(AuthContext);

  useEffect(() => {
    document.title = "Memorable Messages | Home";
  }, []);

  return (
    <div className="home mt-24 grid space-y-10 md:mt-0 md:grid-cols-2 md:gap-x-6 md:space-y-0">
      {/* Hero section */}
      <div className="flex flex-col space-y-6">
        <h1 className="text-3xl font-bold">
          {!token &&
            "Capture life's precious moments with heartfelt messages that last."}
          {token && `Welcome back, ${username}`}
        </h1>
        <p>
          {!token &&
            "Create personalized messages for birthdays, anniversaries, holidays, or any special occasion. Set them to deliver automatically and cherish memories forever."}
          {token &&
            `So excited to see you again, ${username}! We're here to help you make every celebration unforgettable.`}
        </p>
      </div>

      {/* Hero image */}
      <img src={hero} alt="A girl sending wishes" className="md:row-span-2" />

      {/* Call to action */}
      <div className="flex flex-col space-y-6 md:row-start-2 md:row-end-2">
        <h2 className="text-xl">
          {!token &&
            "Sign up for free and start sending memorable messages today!"}
          {token && "Create a new message and bring joy to someone's day!"}
        </h2>
        {!token && (
          <Button asChild className="w-fit">
            <Link to="/signup">Sign Up</Link>
          </Button>
        )}
        {token && (
          <Button asChild className="w-fit">
            <Link to="/occasions">Create</Link>
          </Button>
        )}
      </div>
    </div>
  );
};

export default Home;
