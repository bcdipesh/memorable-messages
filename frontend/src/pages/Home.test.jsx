import React from "react";
import { BrowserRouter } from "react-router-dom";

import { render } from "@testing-library/react";

import Home from "@/pages/Home";

test("Renders without crashing", function () {
  render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );
});

test("Matches snapshot", function () {
  const { asFragment } = render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
