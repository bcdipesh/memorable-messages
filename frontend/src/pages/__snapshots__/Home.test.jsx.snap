// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="home mt-24 grid space-y-10 md:mt-0 md:grid-cols-2 md:gap-x-6 md:space-y-0"
  >
    <div
      class="flex flex-col space-y-6"
    >
      <h1
        class="text-3xl font-bold"
      >
        Capture life's precious moments with heartfelt messages that last.
      </h1>
      <p>
        Create personalized messages for birthdays, anniversaries, holidays, or any special occasion. Set them to deliver automatically and cherish memories forever.
      </p>
    </div>
    <img
      alt="A girl sending wishes"
      class="md:row-span-2"
      src="/src/assets/hero.svg"
    />
    <div
      class="flex flex-col space-y-6 md:row-start-2 md:row-end-2"
    >
      <h2
        class="text-xl"
      >
        Sign up for free and start sending memorable messages today!
      </h2>
      <a
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-fit"
        href="/signup"
      >
        Sign Up
      </a>
    </div>
  </div>
</DocumentFragment>
`;
